envoy/cilium/npds.proto

```diff
@@ -34,62 +34,67 @@ message NetworkPolicy {
   // Required.
   uint64 policy = 1;
 
-  // The part of the policy to be enforced at ingress by the filter.
-  // Optional.
-  DirectionNetworkPolicy ingress = 2;
-
-  // The part of the policy to be enforced at egress by the filter.
-  // Optional.
-  DirectionNetworkPolicy egress = 3;
-}
-
-// A network policy to be enforced on flows either to or from hosts.
-message DirectionNetworkPolicy {
-  // The set of per-port network policies, one per destination L4 port.
-  // Every PortNetworkPolicy element in this set has a unique toPort.
-  // Required and may not be empty.
-  repeated PortNetworkPolicy per_port_policies = 1 [(validate.rules).repeated .min_items = 1];
+  // The part of the policy to be enforced at ingress by the filter, as a set
+  // of per-port network policies, one per destination L4 port.
+  // Every PortNetworkPolicy element in this set has a unique port / protocol
+  // combination.
+  // Optional. If empty, all flows in this direction are denied.
+  repeated PortNetworkPolicy ingress_per_port_policies = 2;
+
+  // The part of the policy to be enforced at egress by the filter, as a set
+  // of per-port network policies, one per destination L4 port.
+  // Every PortNetworkPolicy element in this set has a unique port / protocol
+  // combination.
+  // Optional. If empty, all flows in this direction are denied.
+  repeated PortNetworkPolicy egress_per_port_policies = 3;
 }
 
-// A network policy to be enforced on flows to a specific destination L4 port.
+// A network policy to whitelist flows to a specific destination L4 port,
+// as a conjunction of predicates on L3/L4/L7 flows.
+// If all the predicates of a policy match a flow, the flow is whitelisted.
 message PortNetworkPolicy {
   // The flows' destination L4 port number, as an unsigned 16-bit integer.
-  // Required.
+  // Optional. If not specified, all destination L4 port numbers are matched by
+  // this predicate.
   uint32 port = 1 [(validate.rules).uint32.lte = 65535];
 
   // The flows' L4 transport protocol.
-  // Required.
+  // Optional. If not specified, all L4 transport protocols are matched by this
+  // predicate.
```

Comment:
Again, not sure how this can be optional. Default value (0) seems to be "TCP".

Reply:
Yeah, this can't be wildcarded. Fixed to specify that this is required.